File: singleton/python3/logger.py
# The fact that methods cannot be private in python make the implementation of this design
# patter a bit artificial to my taste. I have added it below nonetheless.

class Logger:
	__instance = None

	def instance():
		if Logger.__instance == None:
			Logger()

		return Logger.__instance


	def __init__(self):
		if Logger.__instance == None:
			Logger.__instance = self
			print("Initializing logger...")
			# Do somthing
			print("Done\n")
		else:
			raise Exception("Trying to instantiate a new instance of Logger.")

	@staticmethod
	def log(message):
		print(message)
		

File: singleton/java/Logger.java
public class Logger {
	private static final Logger instance = new Logger();

	private Logger() {
		System.out.println("Initializing logger...");
		// Do somthing
		System.out.println("Done\n");
	}

	public static Logger getInstance() {
		return instance;
	}

	public void log(String message) {
		System.out.println(message);
	}
}
